drivers/docker/driver.go

```diff
@@ -397,60 +397,26 @@ CREATE:
 	// If the container already exists determine whether it's already
 	// running or if it's dead and needs to be recreated.
 	if strings.Contains(strings.ToLower(createErr.Error()), "container already exists") {
-		containers, err := client.ListContainers(docker.ListContainersOptions{
-			All: true,
-		})
+
+		container, err := d.containerByName(config.Name)
```

Comment:
Note for reviewers: I refactored this big chunk of logic in `createContainer` out to `containerByName` because I wanted to reuse it in `CreateNetwork`. We want callers to be able to determine what to do if the container already existed.